src/pages/index.js

```diff
@@ -0,0 +1,365 @@
+  selectedCardId = data._id;
+  openModal(deleteConfirmationModal);
+}
+
+function handleConfirmDelete() {
+  showLoadingModal("Deleting post...");
+  const originalButtonText = deleteConfirmButton.textContent;
+
+  deleteConfirmButton.textContent = "Deleting...";
+  deleteConfirmButton.disabled = true;
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteConfirmationModal);
+    })
+    .catch((err) => {
+      console.error("Error deleting card:", err);
+      alert("Could not delete card. Please try again later.");
+    })
+    .finally(() => {
+      hideLoadingModal();
+      deleteConfirmButton.textContent = originalButtonText;
+      deleteConfirmButton.disabled = false;
+    });
+}
+
+const handleCancelDelete = () => {
+  console.log("Cancel delete clicked"); // Debug log
+  closeModal(deleteConfirmationModal);
+};
+
+deleteConfirmButton.addEventListener("click", handleConfirmDelete);
+deleteCancelButton.addEventListener("click", handleCancelDelete);
+
+// Define the cardsList variable here
+const cardsList = document.querySelector(".cards__list");
+
+// DOM elements for the New Card Modal
+const cardModal = document.querySelector("#new-card-modal");
+const cardForm = cardModal.querySelector(".modal__form");
+const cardModalCloseBtn = cardModal.querySelector(".modal__close-button");
+const cardSubmitBtn = cardModal.querySelector(".modal__submit-button");
+const cardModalBtn = document.querySelector(".profile__post-button");
+const cardNameInput = cardModal.querySelector("#card-name-input");
+const cardCaptionInput = cardModal.querySelector("#card-caption-input");
+
+function renderCard(data, method = "append") {
+  const cardTemplate = document.querySelector("#card-template").content;
+  const cardElement = cardTemplate.querySelector(".card").cloneNode(true);
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const deleteButton = cardElement.querySelector(".card__del-button");
+  const likeButton = cardElement.querySelector(".card__like-button");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  // Add delete functionality with confirmation
+  deleteButton.addEventListener("click", () => {
+    deleteConfirmationModal.classList.add("modal_opened");
+    handleDeleteCard(cardElement, data);
+  });
+
+  deleteConfirmationModal
+    .querySelector(".modal__close-button")
+    .addEventListener("click", () => {
+      closeModal(deleteConfirmationModal);
+    });
+
+  likeButton.addEventListener("click", () => {
+    likeButton.classList.toggle("card__like-button_liked");
+  });
+
+  cardImageEl.addEventListener("click", () => {
+    const previewModal = document.querySelector("#preview-modal");
+    const modalImage = previewModal.querySelector(".modal__image");
+    const modalCaption = previewModal.querySelector(".modal__caption");
```

Comment:
```js

    const previewModal = document.querySelector("#preview-modal");
    const modalImage = previewModal.querySelector(".modal__image");
    const modalCaption = previewModal.querySelector(".modal__caption");
```

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine